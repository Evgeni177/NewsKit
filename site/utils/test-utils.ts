import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme = renderToFragmentWithThemeFactory(
  docsThemeLight,
);
