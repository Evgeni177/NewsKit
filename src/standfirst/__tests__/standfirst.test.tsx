import {renderToFragmentWithTheme} from '../../test/test-utils';
import {Standfirst} from '..';

describe('Standfirst', () => {
  test('renders with default styles', () => {
    const fragment = renderToFragmentWithTheme(Standfirst);
    expect(fragment).toMatchSnapshot();
  });

  const tags: Array<['h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'span']> = [
    ['h4'],
    ['span'],
  ];
  test.each(tags)('renders as %s html tag', renderStyledTextAs => {
    const fragment = renderToFragmentWithTheme(Standfirst, {
      renderStyledTextAs,
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders with custom style preset', () => {
    const fragment = renderToFragmentWithTheme(Standfirst, {
      overrides: {
        styledText: {
          stylePreset: 'linkPrimary',
        },
      },
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders with custom type preset', () => {
    const fragment = renderToFragmentWithTheme(Standfirst, {
      overrides: {
        styledText: {
          typePreset: 'subhead010',
        },
      },
    });
    expect(fragment).toMatchSnapshot();
  });
});
