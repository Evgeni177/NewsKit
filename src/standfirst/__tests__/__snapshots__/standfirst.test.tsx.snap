// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Standfirst renders as h4 html tag 1`] = `
<DocumentFragment>
  .emotion-0 {
  color: #535353;
  font-family: "Noto Sans",sans-serif;
  font-size: 20px;
  line-height: 1.2;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  padding: 1px 0;
}

.emotion-0::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.25em;
}

.emotion-0::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.3125em;
}

@media screen and (min-width:960px) {
  .emotion-0 {
    font-family: "Noto Sans",sans-serif;
    font-size: 24px;
    line-height: 1.1666666666666667;
    font-weight: 400;
    -webkit-letter-spacing: 0;
    -moz-letter-spacing: 0;
    -ms-letter-spacing: 0;
    letter-spacing: 0;
  }

  .emotion-0::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: calc(-0.2333333333333334em + 1px);
  }

  .emotion-0::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: calc(-0.2958333333333334em + 1px);
  }
}

<h4
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Standfirst renders as span html tag 1`] = `
<DocumentFragment>
  .emotion-0 {
  color: #535353;
  font-family: "Noto Sans",sans-serif;
  font-size: 20px;
  line-height: 1.2;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  padding: 1px 0;
  display: inline-block;
}

.emotion-0::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.25em;
}

.emotion-0::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.3125em;
}

@media screen and (min-width:960px) {
  .emotion-0 {
    font-family: "Noto Sans",sans-serif;
    font-size: 24px;
    line-height: 1.1666666666666667;
    font-weight: 400;
    -webkit-letter-spacing: 0;
    -moz-letter-spacing: 0;
    -ms-letter-spacing: 0;
    letter-spacing: 0;
  }

  .emotion-0::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: calc(-0.2333333333333334em + 1px);
  }

  .emotion-0::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: calc(-0.2958333333333334em + 1px);
  }
}

<span
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Standfirst renders with custom style preset 1`] = `
<DocumentFragment>
  .emotion-0 {
  color: #2F75D7;
  font-family: "Noto Sans",sans-serif;
  font-size: 20px;
  line-height: 1.2;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  padding: 1px 0;
}

.emotion-0::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.25em;
}

.emotion-0::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.3125em;
}

@media screen and (min-width:960px) {
  .emotion-0 {
    font-family: "Noto Sans",sans-serif;
    font-size: 24px;
    line-height: 1.1666666666666667;
    font-weight: 400;
    -webkit-letter-spacing: 0;
    -moz-letter-spacing: 0;
    -ms-letter-spacing: 0;
    letter-spacing: 0;
  }

  .emotion-0::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: calc(-0.2333333333333334em + 1px);
  }

  .emotion-0::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: calc(-0.2958333333333334em + 1px);
  }
}

<h2
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Standfirst renders with custom type preset 1`] = `
<DocumentFragment>
  .emotion-0 {
  color: #535353;
  font-family: "Noto Sans",sans-serif;
  font-size: 16px;
  line-height: 1.25;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  padding: 1px 0;
}

.emotion-0::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.275em;
}

.emotion-0::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.3375em;
}

<h2
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Standfirst renders with default styles 1`] = `
<DocumentFragment>
  .emotion-0 {
  color: #535353;
  font-family: "Noto Sans",sans-serif;
  font-size: 20px;
  line-height: 1.2;
  font-weight: 400;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  padding: 1px 0;
}

.emotion-0::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.25em;
}

.emotion-0::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.3125em;
}

@media screen and (min-width:960px) {
  .emotion-0 {
    font-family: "Noto Sans",sans-serif;
    font-size: 24px;
    line-height: 1.1666666666666667;
    font-weight: 400;
    -webkit-letter-spacing: 0;
    -moz-letter-spacing: 0;
    -ms-letter-spacing: 0;
    letter-spacing: 0;
  }

  .emotion-0::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: calc(-0.2333333333333334em + 1px);
  }

  .emotion-0::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: calc(-0.2958333333333334em + 1px);
  }
}

<h2
    class="emotion-0"
  />
</DocumentFragment>
`;
